src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const stateChanges = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
```

Comment:
In case you don't require the `action` object, you can directly extract the required variables within the `for` condition.
```suggestion
  for (const { type, extraData, keysToRemove } of actions) {
```